src/test/analyze.rs

```diff
@@ -0,0 +1,139 @@
+use crate::{FbasAnalyzer, SolveStatus};
+use batsat::callbacks::{AsyncInterrupt, Basic};
+use std::collections::BTreeMap;
+use std::{io::BufRead, str::FromStr};
+
+#[test]
+fn test_solver_interrupt() -> Result<(), Box<dyn std::error::Error>> {
+    let json_file = std::path::PathBuf::from(
+        "./tests/test_data/random/almost_symmetric_network_16_orgs_delete_prob_factor_3.json",
+    );
+    // first solve it without interruption, it should return `UNSAT`
+    let mut solver = FbasAnalyzer::from_json_path(
+        json_file.as_os_str().to_str().unwrap(),
+        AsyncInterrupt::default(),
+    )?;
+    assert_eq!(solver.solve(), SolveStatus::UNSAT);
+
+    // then we reset it and solve it again, but with interruption
+    let cb = AsyncInterrupt::default();
+    let handle = cb.get_handle();
+    solver = FbasAnalyzer::from_json_path(json_file.as_os_str().to_str().unwrap(), cb)?;
+    std::thread::spawn(move || {
+        std::thread::sleep(std::time::Duration::from_micros(100));
+        handle.interrupt_async();
+    });
+    assert_eq!(solver.solve(), SolveStatus::UNKNOWN);
+    Ok(())
+}
+
+#[test]
+fn test() -> std::io::Result<()> {
+    let expected_results: BTreeMap<&str, SolveStatus> = BTreeMap::from([
+        ("missing_1", SolveStatus::UNSAT),
+        ("circular_2", SolveStatus::UNSAT),
+        ("validators_broken_1", SolveStatus::UNSAT),
+        ("circular_1", SolveStatus::UNSAT),
+        ("top_tier", SolveStatus::UNSAT),
+        (
+            "conflicted_2",
+            SolveStatus::SAT((vec![2.into(), 3.into()], vec![0.into(), 1.into()])),
+        ),
+        ("homedomain_test_1", SolveStatus::UNSAT),
+        (
+            "conflicted_3",
+            SolveStatus::SAT((vec![1.into()], vec![0.into()])),
+        ),
+        (
+            "conflicted",
+            SolveStatus::SAT((vec![1.into(), 2.into()], vec![3.into(), 5.into()])),
+        ),
+    ]);
+
+    for entry in std::fs::read_dir("./tests/test_data/")? {
+        let path = entry?.path();
+        if let Some(extension) = path.extension() {
+            if extension == "json" {
+                let case_name = path.file_stem().unwrap().to_str().unwrap();
+                let expected = expected_results.get(case_name).expect(&format!(
+                    "No expected result found for test case: {}",
+                    case_name
+                ));
+                let mut solver = FbasAnalyzer::from_json_path(
+                    path.as_os_str().to_str().unwrap(),
+                    Basic::default(),
+                )
+                .unwrap();
+                let res = solver.solve();
+
+                match (&res, expected) {
+                    (SolveStatus::SAT((qa, qb)), SolveStatus::SAT((exp_qa, exp_qb))) => {
+                        assert_eq!(qa, exp_qa);
```

Comment:
Checking the quorums are the same may lead to spurious test failures because solvers are in general free to return any model they want.

Reply:
Good point. I've changed all tests to asset on the SAT/UNSAT result and printing out (instead of asserting) the model output.